keanu-project/src/main/java/io/improbable/keanu/vertices/LogProbGraph.java

```diff
@@ -0,0 +1,35 @@
+package io.improbable.keanu.vertices;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+
+public class LogProbGraph {
+
+    private final Map<Vertex<?>, Vertex<?>> inputs;
+    private final DoubleVertex logProbOutput;
+
+    public LogProbGraph(DoubleVertex logProbOutput) {
+        this.inputs = new HashMap<>();
+        this.logProbOutput = logProbOutput;
+    }
+
+    public <T> LogProbGraph addInput(Vertex<T> from, Vertex<T> to) {
+        inputs.put(from, to);
+        return this;
+    }
+
+    public Map<Vertex<?>, Vertex<?>> getInputs() {
```

Comment:
This method is dangerous as the map could be changed by the calling code. It's also unused at the moment - can we remove it?

Reply:
If the inputs map is an ImmutableMap then it's not a problem right? I use it in the TF conversion code, which isn't on this branch so we can drop it until that gets merged in.

Reply:
Right.